modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/RankFeaturesFieldMapper.java

```diff
@@ -117,9 +117,6 @@ public RankFeaturesFieldType fieldType() {
 
     @Override
     public void parse(ParseContext context) throws IOException {
-        if (context.externalValueSet()) {
-            throw new IllegalArgumentException("[rank_features] fields can't be used in multi-fields");
-        }
```

Comment:
I've moved these checks (where they existed) to the TypeParser so they are now caught at mapping time, instead of when you try and index your first document.  There are probably more places where we could do this, but for now I've limited it to just those mappers that were already checking.  I've added tests for them as well.